internal/connect/api.go

```diff
@@ -176,14 +184,45 @@ func updateSystem(body []byte) error {
 	return err
 }
 
+// readUptimeLogFile reads the system uptime log from a given file and
+// returns them as a string array. If the given file does not exist,
+// it will be interpreted as if the system uptime log feature is not
+// enabled. Hence an empty array will be returned.
+func readUptimeLogFile(uptimeLogFilePath string) ([]string, error) {
```

Comment:
A couple of notes. First of all, idiomatically the `uptimeLogFile.Close()` call should be done right after the Open with a `defer` statement. Hence:

```go
uptimeLogFile, err := os.Open(uptimeLogFilePath)
if err != nil {
	return nil, err
}
defer uptimeLogFile.Close()
```

This is important because you are supposed to check for `fileScanner.Err()` right after the `fileScanner.Scan()` loop and return early if there was any error when scanning. That is:

```go
for fileScanner.Scan() {
	logEntries = append(logEntries, fileScanner.Text())
}
if err = fileScanner.Err() {
	return nil, err
}
```

Also, I believe that the call `fileScanner.Split(bufio.ScanLines)` is not needed, no? Isn't splitting by lines the default behavior of Scanner?

Reply:
Good point! Yes, you're right, splitting is done by default.